oasis-reference/src/main/kotlin/com/empowerops/volition/ref_oasis/OptimizerController.kt

```diff
@@ -1,210 +1,293 @@
 package com.empowerops.volition.ref_oasis
 
+import com.google.common.eventbus.EventBus
+import com.google.common.eventbus.Subscribe
 import com.sun.javafx.binding.StringConstant
+import javafx.beans.property.Property
 import javafx.beans.property.SimpleStringProperty
+import javafx.collections.FXCollections
 import javafx.collections.ObservableList
 import javafx.fxml.FXML
 import javafx.scene.control.*
+import javafx.scene.control.cell.TextFieldTreeTableCell
 import javafx.scene.control.cell.TreeItemPropertyValueFactory
 import javafx.scene.input.KeyCode
 import javafx.scene.layout.AnchorPane
 import javafx.scene.layout.VBox
+import javafx.util.converter.DoubleStringConverter
+import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.javafx.JavaFx
 import kotlinx.coroutines.launch
-import tornadofx.selectedItem
+import tornadofx.*
 import java.time.Duration
 
-data class ViewData(
-        val nodes: ObservableList<String>,
-        val allMessages: ObservableList<Message>,
-        val currentEvaluationStatus: SimpleStringProperty,
-        val resultList: ObservableList<Result>,
-        val updateList : ObservableList<String>){
-}
 
 class OptimizerController {
-    @FXML lateinit var view : AnchorPane
-    @FXML lateinit var nodesList : ListView<String>
-    @FXML lateinit var messageTableView : TableView<Message>
-    @FXML lateinit var resultTableView : TableView<Result>
 
-    @FXML lateinit var messageSenderColumn : TableColumn<Message, String>
-    @FXML lateinit var timeColumn : TableColumn<Message, String>
-    @FXML lateinit var messageColumn : TableColumn<Message, String>
-
-    @FXML lateinit var resultSenderColumn : TableColumn<Result, String>
-    @FXML lateinit var typeColumn : TableColumn<Result, String>
-    @FXML lateinit var inputColumn  : TableColumn<Result, String>
-    @FXML lateinit var outputColumn : TableColumn<Result, String>
-
-    @FXML lateinit var paramTreeView : TreeTableView<Parameter>
-    @FXML lateinit var nameColumn : TreeTableColumn<Parameter, String>
-    @FXML lateinit var valueColumn : TreeTableColumn<Parameter, String>
-    @FXML lateinit var lbColumn : TreeTableColumn<Parameter, String>
-    @FXML lateinit var upColumn : TreeTableColumn<Parameter, String>
-
-
-    @FXML lateinit var statusLabel : Label
-    @FXML lateinit var optimizerStatusLabel : Label
-    @FXML lateinit var descriptionLabel : Label
-    @FXML lateinit var selectedNodeInfoBox : VBox
-    @FXML lateinit var timeOutTextField : TextField
-    @FXML lateinit var useTimeout : CheckBox
+    /**
+     * This is backing view object for the tree table view, Not intended to use for any data model
+     */
+    internal class Parameter(
```

Comment:
This cant be private because fxml, i guess the best way to limite it visibility is having proper package for view/controllers with internal